implementations/rust/ockam/ockam_api/src/cli_state/vaults.rs

```diff
@@ -112,6 +174,18 @@ impl CliState {
             None => self.get_or_create_default_named_vault().await,
         }
     }
+
+    /// Move a vault file to another location if the vault is not the default vault
+    /// contained in the main database
+    pub async fn move_vault(&self, vault_name: &str, path: &Path) -> Result<()> {
+        let repository = self.vaults_repository().await?;
+        let vault = self.get_named_vault(vault_name).await?;
+        if vault.path() == self.database_path() {
+            return Err(ockam_core::Error::new(Origin::Api, Kind::Invalid, format!("The vault at path {:?} cannot be moved to {path:?} because this is the default vault", vault.path())).into());
+        };
+        std::fs::rename(vault.path(), path)?;
```

Comment:
Should the process be the following?
1. Copy the file to the new location
2. Update the path in the db
3. Remove the old file

Reply:
Great idea, I updated the code.